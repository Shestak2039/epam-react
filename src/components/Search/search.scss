@import "../../styles/colors";

.search {
  width: 100%;
  margin-right: 30px;

  .search-input {
    width: 100%;
    height: 100%;
    font-size: 20px;
    background-color: $main-theme;
    border: 0;
    color: $default-text-color;
    padding: 10px;
  }
}
