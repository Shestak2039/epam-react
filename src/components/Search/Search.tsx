import React from 'react';

import './search.scss';

export interface SearchProps {
    searchQuery: string;
    setSearchQuery: (value: string) => void;
}

const Search: React.FunctionComponent<SearchProps> = ({searchQuery, setSearchQuery}) => (
    <div className="search">
        <input
            name="search-input"
            id="search-input"
            className="search-input"
            placeholder="What do you want to watch?"
            value={searchQuery} onChange={(event) => setSearchQuery(event.target.value)}/>
    </div>
);

export default Search;
