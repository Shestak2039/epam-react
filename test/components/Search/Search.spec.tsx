import React from 'react';
import { render } from '@testing-library/react';
import Search from '../../../src/components/Search/Search';
import userEvent from '@testing-library/user-event/dist';

describe('NoMatch', () => {
    test('renders NoMatch snapshot', () => {
        const mockSearchQuery = 'mockSearchQuery';
        const mockSetSearchQuery = jest.fn();
        
        const { asFragment, getByPlaceholderText } = render(<Search searchQuery={mockSearchQuery} setSearchQuery={mockSetSearchQuery}/>);

        userEvent.type(getByPlaceholderText('What do you want to watch?'), mockSearchQuery);
        
        expect(asFragment()).toMatchSnapshot();
    });
});
