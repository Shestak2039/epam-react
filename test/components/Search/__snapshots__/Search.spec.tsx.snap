// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch renders NoMatch snapshot 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <input
      class="search-input"
      id="search-input"
      name="search-input"
      placeholder="What do you want to watch?"
      value="mockSearchQuery"
    />
  </div>
</DocumentFragment>
`;
